File: 3483-alternating-groups-ii/3483-alternating-groups-ii.cpp
class Solution {
public:
    int numberOfAlternatingGroups(vector<int>& colors, int k) {
        int n = colors.size();
        for(int i = 0 ;i < k;i++)colors.push_back(colors[i]);
        deque<int>sameIndx;int answer = 0;
        for(int i = 0;i<k-1;i++){
            if(colors[i] == colors[i+1]){
                sameIndx.push_back(i);
                //cout<<"i= "<<i<<" ";
            }
        }//[0, 1, 0, 0,1,0, 1, 0 ,1, 0, 0,1]
        //  0, 1, 2, 3,4,5, 6, 7 ,8, 9, 10,11
        if(sameIndx.empty())answer++;
        //  for(int i = 0 ;i<n+k-1;i++){
        //     cout<<colors[i]<<" ";
        //  }cout<<endl;
        for(int i = 0,j=k;j <n+k-1;j++,i++){
           // cout<<i<<" "<<j<< " "<<sameIndx.size()<<endl;
            if(colors[j] == colors[j-1]){
                sameIndx.push_back(j-1);
              //  cout<<"j= "<<j<<" ";
            }
            if(!sameIndx.empty() && sameIndx.front() == i){
                sameIndx.pop_front();
            }
            if(sameIndx.empty())
            answer++;
            
        }

        return answer;
    }
};

File: 3483-alternating-groups-ii/3483-alternating-groups-ii.cs
public class Solution {
    public int NumberOfAlternatingGroups(int[] colors, int k) {
        int answer = 0;
        int n = colors.Length;
        List<int> v = new List<int>();
        PriorityQueue<int,int> pq = new PriorityQueue<int,int>();
        foreach(var u in colors) v.Add(u);
        for(int i = 0 ;i < k-1;i++) v.Add(colors[i]);
        n = v.Count;
       // Console.WriteLine(n);
        for(int i = 0 ; i < k-1;i++){
            if(v[i] == v[i+1]){
                pq.Enqueue(i,i);
               // Console.Write(" "+i);
            }
        }
        if(pq.Count == 0) answer++;
        for(int i = 0, j = k;j<n;j++,i++){
            if(v[j] == v[j-1]){
                pq.Enqueue(j-1,j-1);
                //Console.Write(" "+j);
            }
            if(pq.Count!=0 && pq.Peek() == i){
                pq.Dequeue();
            }
            if(pq.Count == 0)answer++;
        }
        return answer;
    }
}